platform/linux-generic/odp_queue_basic.c

```diff
@@ -1017,13 +1017,15 @@ static int queue_info(odp_queue_t handle, odp_queue_info_t *info)
 	if (queue->type == ODP_QUEUE_TYPE_AGGR) {
 		queue_entry_t *base_queue = qentry_from_handle(queue->aggr.base_queue);
 
+		memset(info, 0, sizeof(*info));
 		info->name = base_queue->name;
+		info->type = queue->type;
 		info->param = base_queue->param;
-
-		/* Override some of the base queue parameters */
-		info->param.type = ODP_QUEUE_TYPE_AGGR;
-		info->param.context = NULL;
-		info->param.context_len = 0;
+		info->param.aggr = NULL;
+		info->aggr_config.pool = queue->aggr.pool;
+		info->aggr_config.max_tmo_ns = 0;
+		info->aggr_config.max_size = queue->aggr.max_size;
+		info->aggr_config.event_type = queue->aggr.event_type;
```

Comment:
The non-aggregator info collection is done while locked, I guess this should be as well.